torch/_dynamo/output_graph.py

```diff
@@ -87,6 +87,7 @@ class OutputGraphState(NamedTuple):
     param_name_to_source: Optional[Dict[str, Source]]
     side_effects: SideEffects
     timestamp: int
+    tensor_id_to_fake_clone: Dict[int, torch._subclasses.FakeTensor]
```

Comment:
Can we please use a weakref instead of an id. This will help prevent awful bugs where the input Tensor is not kept live and then an id gets reused.

Reply:
torch/utils/weak.py has the stuff you want

Reply:
sg!